### PR TITLE
2pc: cleanup primary key first

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,
     This PR fix the following bug:
     When commit primary receive a undetermined error,  we can not sure whether the `txn` had been committed successfully. On the master branch,  we would do clean up parallel when meet the undetermined error ,  while the primary would be failed with `txn has been committed` and the secondaries would be cleaned up -- we would lost secondaries data in this case. 

In this PR, we do two changes to fix the bug:
1.  For the `clean up` operation, always clean up primary key first.
2.  When a undetermined error received by the `commit` operation, do not cleanup any thing.
@disksing @hhkbp2 PTAL

### CODE REVIEW

**Comment by AndreMouche** [comment]
@siddontang @disksing @hhkbp2 @zhangjinpeng1987 PTAL

**Comment by disksing** [comment]
LGTM.

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -206,8 +207,8 @@ func (c *twoPhaseCommitter) doActionOnKeys(bo *Backoffer, action twoPhaseCommitA
 	}
 
 	firstIsPrimary := bytes.Equal(keys[0], c.primary())
-	if firstIsPrimary && action == actionCommit {
-		// primary should be committed first.
+	if firstIsPrimary && action != actionPrewrite {
```
Comment (hhkbp2) [inline store/tikv/2pc.go]: IMO it's clearer to write `!=` like
```
action == actionCommit || action == actionCleanup
```

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -368,7 +369,13 @@ func (c *twoPhaseCommitter) prewriteSingleBatch(bo *Backoffer, batch batchKeys)
 			// We need to cleanup all written keys if transaction aborts.
 			c.mu.Lock()
 			defer c.mu.Unlock()
-			c.mu.writtenKeys = append(c.mu.writtenKeys, batch.keys...)
+			if bytes.Equal(batch.keys[0], c.primary()) {
```
Comment (hhkbp2) [inline store/tikv/2pc.go]: A comment to decribe why primary key needs to be appended ahead is appreciated.

**Comment by hhkbp2** [comment]
Rest LGTM

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**